scipy/sparse/_compressed.py

```diff
@@ -124,14 +125,42 @@ def _getnnz(self, axis=None):
             axis, _ = self._swap((axis, 1 - axis))
             _, N = self._swap(self.shape)
             if axis == 0:
-                return np.bincount(downcast_intp_index(self.indices),
-                                   minlength=N)
+                return np.bincount(downcast_intp_index(self.indices), minlength=N)
             elif axis == 1:
                 return np.diff(self.indptr)
             raise ValueError('axis out of bounds')
 
     _getnnz.__doc__ = _spbase._getnnz.__doc__
 
+    def count_nonzero(self, axis=None):
+        self.sum_duplicates()
```

Comment:
This may be a blocker: historically, we haven't done silent mutation for the `count_nonzero()` method, and user code may break if we start.